fe/fe-core/src/main/java/org/apache/doris/job/extensions/mtmv/MTMVJob.java

```diff
@@ -140,27 +140,37 @@ public List<MTMVTask> createTasks(TaskType taskType, MTMVTaskContext taskContext
 
     /**
      * if user trigger, return true
-     * if system trigger, Check if there are any system triggered tasks, and if so, return false
+     * else, only can have 2 task. because every task can refresh all data.
      *
      * @param taskContext
      * @return
      */
     @Override
     public boolean isReadyForScheduling(MTMVTaskContext taskContext) {
-        if (taskContext != null) {
+        if (isManual(taskContext)) {
             return true;
         }
         List<MTMVTask> runningTasks = getRunningTasks();
+        int runningNum = 0;
         for (MTMVTask task : runningTasks) {
-            if (task.getTaskContext() == null || task.getTaskContext().getTriggerMode() == MTMVTaskTriggerMode.SYSTEM) {
-                LOG.warn("isReadyForScheduling return false, because current taskContext is null, exist task: {}",
-                        task);
-                return false;
+            if (!isManual(task.getTaskContext())) {
+                runningNum++;
+                // TODO: 2024/5/9 When the scheduling framework supports running only one task simultaneously,
+                //  change to 2 to prevent message loss
```

Comment:
cannot understand this TODO. could u explain it clearly? why change to 2 could  prevent message loss